typeclasses/characters.py

```diff
@@ -91,6 +75,71 @@ def str(self, amount):
 
         self.db.slots = {}
 
+    # helper method, checks if stat is valid
+    def find_stat(self, stat):
+        if stat in self.db.primary_traits:
+            return self.db.primary_traits[stat]
+        elif stat in self.db.secondary_traits:
+            return self.db.secondary_traits[stat]
+        else:
+            return None
+
+    # helper method, checks if race gets extra language points
+    def determine_language_points(self):
+        if self.bonuses['languages']:
+            return self.bonuses['languages']
+
+    def base_stat(self, stat, amount):
+        # sets the secondary traits
+        if stat == 'vitality':
+            base_stat('health', amount)
```

Comment:
Using base_stat within base_stat is recursive functionality. Python does not do exceptionally well with recursion, and iteration is preferred.

This should look something like this instead:

```python
if stat == 'vitality':
   for secondary in ['health', 'stamina', 'fortitude', 'reflex']:
      secondary.base = amount
```